modules/asm/main.tf

```diff
@@ -14,95 +14,76 @@
+  gke_release_channel_fixed = local.gke_release_channel == "UNSPECIFIED" ? "" : local.gke_release_channel
+  // In order or precedence, use (1) user specified channel, (2) GKE release channel, and (3) regular channel
+  channel          = lower(coalesce(var.channel, local.gke_release_channel_fixed, "regular"))
+  revision_name    = "asm-managed${local.channel == "regular" ? "" : "-${local.channel}"}"
+  mesh_config_name = "istio-${local.revision_name}"
+  // CNI should be enabled if either enable_cni or enable_mdp are set
+  enable_cni = var.enable_cni || var.enable_mdp
 }
 
-resource "google_project_iam_member" "asm_iam" {
-  for_each = toset(local.asm_iam_member != "" ? (var.managed_control_plane ? local.mcp_iam_roles : local.iam_roles) : [])
+data "google_container_cluster" "asm_cluster" {
   project  = var.project_id
-  role     = each.value
-  member   = "serviceAccount:${local.asm_iam_member}"
+  name     = var.cluster_name
+  location = var.cluster_location
 }
 
-module "asm-services" {
-  source  = "terraform-google-modules/project-factory/google//modules/project_services"
-  version = "~> 11.3"
+resource "kubernetes_namespace" "system_namespace" {
+  metadata {
+    name = "istio-system"
+  }
+}
 
-  count = var.enable_gcp_apis ? 1 : 0
+resource "kubernetes_config_map" "mesh_config" {
+  count = length(var.mesh_config) == 0 ? 0 : 1
 
-  project_id                  = var.project_id
-  disable_services_on_destroy = false
-  disable_dependent_services  = false
+  metadata {
+    name      = local.mesh_config_name
+    namespace = kubernetes_namespace.system_namespace.metadata[0].name
+    labels = {
+      "istio.io/rev" = local.revision_name
+    }
+  }
+
+  data = {
+    mesh = yamlencode(var.mesh_config)
+  }
+}
 
-  # https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/1cf61b679cd369f42a0e735f8e201de1a6a6433b/scripts/asm-installer/install_asm#L2005
-  activate_apis = [
-    "container.googleapis.com",
-    "monitoring.googleapis.com",
-    "logging.googleapis.com",
-    "cloudtrace.googleapis.com",
-    "meshtelemetry.googleapis.com",
-    "meshconfig.googleapis.com",
-    "meshca.googleapis.com",
-    "iamcredentials.googleapis.com",
-    "gkeconnect.googleapis.com",
-    "gkehub.googleapis.com",
-    "cloudresourcemanager.googleapis.com",
-    "stackdriver.googleapis.com",
-  ]
+resource "kubernetes_config_map" "asm_options" {
+  metadata {
+    name      = "asm-options"
+    namespace = kubernetes_namespace.system_namespace.metadata[0].name
 }
 
-module "asm_install" {
-  source  = "terraform-google-modules/gcloud/google//modules/kubectl-wrapper"
-  version = "~> 3.1"
 
-  module_depends_on = concat([var.cluster_endpoint], local.additional_depends_on)
+  data = {
+    CROSS_CLUSTER_SERVICE_DISCOVERY = var.enable_cross_cluster_service_discovery ? "ON" : "OFF"
+  }
+}
+
+# TODO(Monkeyanator) due to a bug in the gke_hub_feature resource implementation this fails when enabling an enabled
```

Comment:
Is there an upstream bug filed?

Reply:
Not yet, will update here once created.

Reply:
I'm unable to reproduce the bug, added the feature enablement back here.